src/SonarScanner.MSBuild.PreProcessor/Interfaces/IAnalyzerProvider.cs

```diff
@@ -29,9 +29,8 @@ public interface IAnalyzerProvider
     /// <summary>
     /// Sets up a Roslyn analyzer to run as part of the build
     /// i.e. creates the Roslyn ruleset and provisions the analyzer's assemblies
-    /// and rule parameter files
+    /// and rule parameter files.
```

Comment:
This interface is implemented by a single class => it just makes the code difficult to navigate. Throw it away